Stocksnap test takes 40s to run

## Description
<!-- Concisely describe the bug. Compare your experience with what you expected to happen. -->
<!-- For example: "I clicked the 'submit' button and instead of seeing a thank you message, I saw a blank page." -->
While running tests recently, I realized that some tests were taking an incredibly long time. After running `just test --durations=5`, I got the following output:

```
=============================================================================================== slowest 5 durations ================================================================================================
39.04s call     tests/dags/providers/provider_api_scripts/test_stocksnap.py::test_process_batch
2.00s call     tests/dags/common/test_requester.py::test_get_delays_processing
1.03s call     tests/dags/common/loader/test_sql.py::test_upsert_records_replaces_updated_on_and_last_synced_with_source
0.42s call     tests/dags/common/loader/test_sql.py::test_loaders_load_good_tsv[_load_s3_tsv]
0.25s call     tests/dags/common/sensors/test_single_run_external_dags_sensor.py::TestExternalDAGsSensor::test_fails_if_external_dag_does_not_exist
```

This seems to be a recent development, possibly related to #601

## Resolution
<!-- Replace the [ ] with [x] to check the box. -->
- [ ] 🙋 I would be interested in resolving this bug.

Ah, it looks like this test is actually making requests and pytest-socket isn't correctly capturing that in the current runs:

```

pytest -n 1 tests/dags/providers/provider_api_scripts/test_stocksnap.py::test_process_batch
/usr/local/airflow/.local/lib/python3.10/site-packages/pytest_sugar.py:168: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
  from distutils.version import LooseVersion
Test session starts (platform: linux, Python 3.10.1, pytest 6.2.5, pytest-sugar 0.9.4)
rootdir: /usr/local/airflow, configfile: pytest.ini
plugins: xdist-2.5.0, socket-0.5.1, flaky-3.7.0, forked-1.4.0, sugar-0.9.4, anyio-3.5.0, raises-0.11, mock-3.6.1
gw0 Ccollecting ... /usr/local/airflow/.local/lib/python3.10/site-packages/pytest_sugar.py:168: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
  from distutils.version import LooseVersion
gw0 [1]


―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― test_process_batch ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
[gw0] linux -- Python 3.10.1 /usr/local/bin/python

====large traceback redacted====

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'pytest_socket.disable_socket.<locals>.GuardedSocket'>, family = <AddressFamily.AF_INET: 2>, type = <SocketKind.SOCK_STREAM: 1>, proto = 6, fileno = None

    def __new__(cls, family=-1, type=-1, proto=-1, fileno=None):
        if _is_unix_socket(family) and allow_unix_socket:
            return super().__new__(cls, family, type, proto, fileno)
    
>       raise SocketBlockedError()
E       pytest_socket.SocketBlockedError: A test tried to use socket.socket.

.local/lib/python3.10/site-packages/pytest_socket.py:80: SocketBlockedError
----------------------------------------------------------------------------------------------- Captured stdout call -----------------------------------------------------------------------------------------------
[2022-07-21 23:59:57,688] {stocksnap.py:76} INFO - STARTING
[2022-07-21 23:59:57,700] {requester.py:55} ERROR - Error with the request for URL: https://cdn.stocksnap.io/img-thumbs/960w/OAJ3645ZWF.jpg.
[2022-07-21 23:59:57,700] {requester.py:56} INFO - SocketBlockedError: A test tried to use socket.socket.
[2022-07-21 23:59:57,700] {requester.py:57} INFO - Using query parameters None
[2022-07-21 23:59:57,700] {requester.py:58} INFO - Using headers None
[2022-07-21 23:59:57,700] {stocksnap.py:102} INFO - ENDING
------------------------------------------------------------------------------------------------ Captured log call -------------------------------------------------------------------------------------------------
INFO     providers.provider_api_scripts.stocksnap:stocksnap.py:76 STARTING
ERROR    common.requester:requester.py:55 Error with the request for URL: https://cdn.stocksnap.io/img-thumbs/960w/OAJ3645ZWF.jpg.
INFO     common.requester:requester.py:56 SocketBlockedError: A test tried to use socket.socket.
INFO     common.requester:requester.py:57 Using query parameters None
INFO     common.requester:requester.py:58 Using headers None
INFO     providers.provider_api_scripts.stocksnap:stocksnap.py:102 ENDING

 tests/dags/providers/provider_api_scripts/test_stocksnap.py ⨯                                                                                                                                       100% ██████████
============================================================================================= short test summary info ==============================================================================================
FAILED tests/dags/providers/provider_api_scripts/test_stocksnap.py::test_process_batch - pytest_socket.SocketBlockedError: A test tried to use socket.socket.

Results (1.45s):
       1 failed
         - tests/dags/providers/provider_api_scripts/test_stocksnap.py:105 test_process_batch
```

--- a/openverse_catalog/dags/common/requester.py
+++ b/openverse_catalog/dags/common/requester.py
@@ -5,6 +5,16 @@ import oauth2
 import requests
 
 
+# pytest_socket will not be available in production, so we must create a shim for
+# that specific exception so this code does not error out.
+try:
+    from pytest_socket import SocketConnectBlockedError
+except ImportError:
+
+    class SocketConnectBlockedError(Exception):
+        pass
+
+
 logger = logging.getLogger(__name__)
 
 
@@ -51,6 +61,10 @@ class DelayedRequester:
                     f"Status code: {response.status_code}"
                 )
             return response
+        except SocketConnectBlockedError:
+            # This exception will only be raised during testing, and it *must*
+            # be re-raised and bubbled up the stack
+            raise
         except Exception as e:
             logger.error(f"Error with the request for URL: {url}.")
             logger.info(f"{type(e).__name__}: {e}")

